src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -151,20 +153,24 @@ public MessageWrapper fromProtoMessage(Message receivedMessage, boolean useDicti
         return messageWrapper;
     }
 
-    public IMessage fromProtoFilter(MessageFilter messageFilter, String messageName) {
+    public IMessage fromProtoFilter(MessageFilter messageFilter, FilterSettings filterSettings, String messageName) {
         if (logger.isDebugEnabled()) {
             logger.debug("Converting filter {} as {}", shortDebugString(messageFilter), messageName);
         }
         IMessage message = messageFactory.createMessage(dictionaryURI, messageName);
         for (Entry<String, ValueFilter> filterEntry : messageFilter.getFieldsMap().entrySet()) {
-            message.addField(filterEntry.getKey(), traverseFilterField(filterEntry.getKey(), filterEntry.getValue()));
+            message.addField(filterEntry.getKey(), traverseFilterField(filterEntry.getKey(), filterEntry.getValue(), filterSettings));
         }
 
         logger.debug("Filter '{}' converted {}", messageName, message);
         return message;
     }
 
-    public IMessage fromMetadataFilter(MetadataFilter filter, String messageName) {
+    public IMessage fromProtoFilter(MessageFilter messageFilter, String messageName) {
+        return fromProtoFilter(messageFilter, null, messageName);
```

Comment:
I think we should use the default filter setting instead of `null`. They should have both precisions equal to zero.